### PR TITLE
[Review] Scalar types in Reviews

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | https://github.com/Sylius/Sylius/issues/8425 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Core/Model/Product.php [hunk 1]
```diff
@@ -268,31 +268,31 @@ public function getAcceptedReviews()
...
+    public function removeReview(ReviewInterface $review): void
     {
         $this->reviews->remove($review);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getAverageRating()
+    public function getAverageRating(): float
```
Comment (pamil) [inline src/Sylius/Component/Core/Model/Product.php]: `?float`, isn't it? :)

File: src/Sylius/Component/Review/Factory/ReviewFactory.php [hunk 2]
```diff
@@ -57,8 +59,9 @@ public function createForSubject(ReviewableInterface $subject)
     /**
      * {@inheritdoc}
      */
-    public function createForSubjectWithReviewer(ReviewableInterface $subject, ReviewerInterface $reviewer = null)
+    public function createForSubjectWithReviewer(ReviewableInterface $subject, ?ReviewerInterface $reviewer = null): ReviewInterface
```
Comment (pamil) [inline src/Sylius/Component/Review/Factory/ReviewFactory.php]: We can remove the default `null` value here, when called without the second parameter then `createForSubject` should be used instead.

File: src/Sylius/Component/Review/Factory/ReviewFactoryInterface.php [hunk 3]
```diff
@@ -28,13 +28,13 @@
...
+    public function createForSubject(ReviewableInterface $subject): ReviewInterface;
 
     /**
      * @param ReviewableInterface $subject
      * @param ReviewerInterface|null $reviewer
      *
      * @return ReviewInterface
      */
-    public function createForSubjectWithReviewer(ReviewableInterface $subject, ReviewerInterface $reviewer = null);
+    public function createForSubjectWithReviewer(ReviewableInterface $subject, ?ReviewerInterface $reviewer): ReviewInterface;
```
Comment (Zales0123) [inline src/Sylius/Component/Review/Factory/ReviewFactoryInterface.php]: Unfortunately it's required to have it here for guest reviews. I've just made it required to pass the `$reviewer` explicitly as the default value was ridiculous.

**Comment by pamil** [comment]
Thanks Matt! ☂️

**Pull request merged**